src/arch/x86_64/mod.rs

```diff
@@ -80,19 +86,76 @@ pub fn output_message_byte(byte: u8) {
 	unsafe { COM1.send(byte) };
 }
 
+// Right now, the kernel binary has to be hardcoded into the loader.
+// The binary has to be in the same directory (or its whereabouts have to specified in the "include_bytes!" statement).
 #[cfg(target_os = "uefi")]
 pub unsafe fn find_kernel() -> &'static [u8] {
-	&[1, 2, 3]
+	include_bytes!("hermit-rs-template")
 }
 
+/// This is the actual boot function.
+/// The bootstack is cleared and provided/calculated BOOT_INFO is written into before the actual call to the assembly code to jump into the kernel.
 #[cfg(target_os = "uefi")]
 pub unsafe fn boot_kernel(
-	_elf_address: Option<u64>,
-	_virtual_address: u64,
-	_mem_size: u64,
-	_entry_point: u64,
+	rsdp_addr: u64,
+	kernel_addr: u64,
+	filesize: usize,
+	kernel_info: LoadedKernel,
+	runtime_system_table: uefi::table::SystemTable<uefi::table::Runtime>,
+	start_address: usize,
+	end_address: usize,
 ) -> ! {
```

Comment:
We might want to move this and the `target_os = "none"` one into separate modules. Let's coordinate on this. :)